replication/README.md

```diff
@@ -312,3 +317,36 @@ message ResyncVolumeResponse{
 | Not authenticated                                | 16 UNAUTHENTICATED    | The invoked RPC does not carry secrets that are valid for authentication.                                                                                                                                                                                                                                                                                                                                                                                                                                                                 | Caller SHALL either fix the secrets provided in the RPC, or otherwise regalvanize said secrets such that they will pass authentication by the Plugin for the attempted RPC, after which point the caller MAY retry the attempted RPC. |
 | Error is Unknown                                 | 2 UNKNOWN             | Indicates that a unknown error is generated                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               | Caller MUST study the logs before retrying                                                                                                                                                                                            |
+
+### GetVolumeReplicationInfo
+
+```protobuf
+// GetVolumeReplicationInfoRequest holds the required information to get
+// the Volume replication information.
+message GetVolumeReplicationInfoRequest {
+  // The identifier for this volume, generated by the plugin during
+  // CreateVolume CSI RPC call.
+  // This field is REQUIRED.
+  // This field MUST contain enough information to uniquely identify
+  // this specific volume vs all other volumes supported by this plugin.
+  // This field SHALL be used by the CO in subsequent calls to refer to
+  // this volume.
+  string volume_id = 1;
+  // Secrets required by the plugin to complete the request.
+  map<string, string> secrets = 2 [(csi.v1.csi_secret) = true];
```

Comment:
As other calls like `ResyncVolumeRequest` contain the `parameters` map, it seems appropriate to pass there here as well. I imagine the parameters can contain things like hostnames of storage systems that are needed to obtain the volume replication information?

What is the reason to include it in `ResyncVolumeRequest`, but not here?

Reply:
Not sure about it, but didn't find any specific use of it.  Maybe if @Madhu-1 can answer?

Reply:
For actual set operations, you need parameters to set it on the resource, during get operation, we don't need any parameters. a volume_id should be enough to get the resource details.

Reply:
I do not understand what the difference between set/get are related to connecting to the storage provider. Even set-operations do not include parameters for the operation, only for the connection?

Could you explain with an example what parameters Promote/Demote/Resync might require, that are not needed for `GetVolumeReplicationInfo`? From my understanding, all these procedures are executed pretty much the same...

Reply:
from the SPEC point of view, if you have the volumeID and the secret, you should be able to connect to the storage provider and do whatever option you want on the volume.
for Promote/Demote/Resync, we can pass any other storage provider-specific parameters for this function, the storage provide can take any action on it example we can have a force option where the storage provider can do some force promote/force demote etc.
For the Get option where do you think a parameter is required?

Reply:
Ok, I guess that makes sense. Thanks